recipes/repo_hosting.rb

```diff
@@ -0,0 +1,107 @@
+#
+# Cookbook Name:: phabricator
+# Recipe:: repo_hosting
+#
+# Copyright 2014, MET Norway
+#
+# Authors: Kim Tore Jensen <kimtj@met.no>,
+#          Andrew Mulholland <andrew@bash.sh>
+#
+# Sets up phabricator for hosting repositories over SSH
+
+# To prevent the `vcsuser` (i.e. git ) user from being locked, a password needs to be provided with
+# account creation. Using OpenSSLCookbook's RandomPassword function to generate a secure password.
+unless node['phabricator']['vcspassword']
+    Chef::Recipe.send(:include, OpenSSLCookbook::RandomPassword)
+    require 'digest/sha2'
+    node.set['phabricator']['vcspassword'] = Digest::SHA512.hexdigest(random_password(length: 50))
+end
+
+user node['phabricator']['vcsuser'] do
+    comment 'VCS User'
+    home File.join('/home', node['phabricator']['vcsuser'])
```

Comment:
This variable should definitely be configured in node attributes, instead of auto-generated.